File: sdks/csharp/rest/src/Brume.Rest/Types/PublishResponse.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

/// <summary>
/// `POST /v1/channels/:channel/publish` success response.
/// </summary>
[Serializable]
public record PublishResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("channel")]
    public required string Channel { get; set; }

    [JsonPropertyName("event")]
    public required string Event { get; set; }

    /// <summary>
    /// Number of subscribers the message was fanned out to.
    /// </summary>
    [JsonPropertyName("recipients")]
    public required int Recipients { get; set; }

    /// <summary>
    /// Always `"published"`. Reserved for future state changes.
    /// </summary>
    [JsonPropertyName("status")]
    public required string Status { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Types/PublishResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;

/**
 * `POST /v1/channels/:channel/publish` success response.
 */
class PublishResponse extends JsonSerializableType
{
    /**
     * @var string $channel
     */
    #[JsonProperty('channel')]
    public string $channel;

    /**
     * @var string $event
     */
    #[JsonProperty('event')]
    public string $event;

    /**
     * @var int $recipients Number of subscribers the message was fanned out to.
     */
    #[JsonProperty('recipients')]
    public int $recipients;

    /**
     * @var string $status Always `"published"`. Reserved for future state changes.
     */
    #[JsonProperty('status')]
    public string $status;

    /**
     * @param array{
     *   channel: string,
     *   event: string,
     *   recipients: int,
     *   status: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->channel = $values['channel'];
        $this->event = $values['event'];
        $this->recipients = $values['recipients'];
        $this->status = $values['status'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
